### DISCUSSION TITLE
Verbose output from before/after timing report dumps

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#1628] [General]
In the synth, floorplan, place, cts, route stages there are before/after timing reports dumped into the stdout/err.

This makes logs very verbose.

I'm not quite sure where else one would put this information, but perhaps reports/timing.txt?

Thoughts?

**Comment by maliberty** [comment]
We don't currently support redirection so there is no option at this point to support this.

**Reply by oharboe** [reply]
I see. That aside, thoughts on this workflow?

**Reply by maliberty** [reply]
If we had such a feature then I would be fine to put the timing reports in a separate file.

**Comment by maliberty** [comment]
I have to correct my previous answer.  I checked and the sta redirect (>) is working.  I guess it got resolved at some point and I didn't notice.  So a change like
```
-  report_checks -path_delay min -fields {slew cap input nets fanout} -format full_clock_expanded
+  report_checks -path_delay min -fields {slew cap input nets fanout} -format full_clock_expanded > $::env(REPORTS_DIR)/$when-min.rpt
```
appears to work.

**Reply by oharboe** [reply]
So no it is a matter of finding an updated policy that is better for everyone...

About the reports: isnt it the difference between timing before and after that is interesting?

Could this difference be articulated tersely enough that it  could go into the main log and then the full reports to the reports dir in case one wants to dig deeper?

Or perhaps opening the openroad GUI is preferable more often than looking at the full reports and the full reports are not really needed?

**Reply by maliberty** [reply]
I think the reports are useful but putting them to the side of the main log makes sense.  We already print a bunch of summary stats (tns/wns) that can stay in the log.
